### DISCUSSION TITLE
Change Row Background Color Based on Specific column value

### DISCUSSION THREAD

**Discussion by Hegabovic** [filamentphp/filament#10671] [Help] [answered]
i'm trying to change the row background color based on true and false values 

```
 protected function getTableRecordClassesUsing(): ?Closure
    {
        return function (Booking $record) {
            return match ($record->is_paid) {
                true => 'opacity-30',
                false => [
                    'border-l-solid',
                    'border-l-2',
                    'border-l-orange-600',
                    'dark:border-l-orange-300' => config('filament.dark_mode'),
                    'opacity-30',
                ],
                default => null,
            };
        };
    }

**Comment by leandrocfe** [comment, marked as answer]
Use `recordClasses`
https://filamentphp.com/docs/3.x/tables/advanced#custom-row-classes

If you are using custom colors, you should create a custom theme
https://filamentphp.com/docs/3.x/panels/themes#creating-a-custom-theme

*These classes are not automatically compiled by Tailwind CSS. If you want to apply Tailwind CSS classes that are not already used in Blade files, you should update your content configuration in tailwind.config.js to also scan for classes inside your directory: './app/Filament/**/*.php'*

**Comment by yakou32** [comment]
@leandrocfe 
Hey!

This is exactly  what I'm trying to do:
```
return $table
            ->recordClasses(fn (Model $record) => $record->amount > 0 ? 'bg-lime-700' : null)
```

And I added the directory in `tailwind.config.js`:
```
    content: [
        "./vendor/laravel/framework/src/Illuminate/Pagination/resources/views/*.blade.php",
        "./storage/framework/views/*.php",
        "./resources/views/**/*.blade.php",
        "./app/Filament/**/*.php",
    ],
```` 

Tailwind class is added to the HTML, however, no rendering appears.

What did I miss?

Thanks :)

**Reply by edrisaturay** [reply]
same with me.. what did I miss

**Reply by leandrocfe** [reply]
Make sure your content configuration in tailwind.config.js has your directory/class

I have a tutorial in pt-BR
https://www.youtube.com/watch?v=7SnzOjPV7ms

**Reply by mgkimsal** [reply]
@leandrocfe - your video is showing creating an entirely new theme.  I think some of us would want to do this without having to create a new theme.  The documentation seems to indicate this can be done without creating an entirely new theme, but I, like apparently many others, seem to not be able to make this work by simply adding the ./app/Filament/** path to various config files and re-running.

Any other pointers?

**Reply by leandrocfe** [reply]
No, you must create a new theme to recompile the css, if you are using Tailwind CSS classes that are not already used in Filament

**Reply by renky** [reply]
You can just add your custom classes like this:
Add css in your app service provider

```php
        FilamentAsset::register([
            Css::make('custom-stylesheet', __DIR__ . '/../../resources/css/filament.css'),
        ]);
```

When you now publish your filament assets:
```bash
php artisan filament:assets
```
then it is automatically added to all your filament pages.


now you can simply add the classes of your custom-stylesheet to your tables/records and everything else...

if your css file contains a class like ".my-lime-bg" with your color settings, you can just add it to your table
```php
    return $table
            ->recordClasses(fn (Model $record) => $record->amount > 0 ? 'my-lime-bg' : null)
```

all records have your class now and your styles are active then without creating an own theme.

You have only to keep in mind that tailwind isn't executed on the filament.css... if you want to use tailwind additionally it gets more komplex, you have to care about the generated filename and more... Wasn't a solution in my case - i just needed a special border-setting and thats what I added to my filament.css in plain old css (with !important, because otherwise the filament default theme settings won)...
